src/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+)
+
+type logger interface {
+	log(format string, args ...interface{}) error
+}
+
+type checker interface {
+	run(fset *token.FileSet, logger logger) error
+}
+
+type consoleLogger struct{}
+
+func (c consoleLogger) log(format string, args ...interface{}) error {
+	_, err := fmt.Printf(format, args...)
+	return err
+}
+
+var checkers = []checker{}
+
+func blame(format string, args ...interface{}) {
+	log.Printf(format, args...)
+	flag.Usage()
+	os.Exit(1)
+}
+
+func main() {
+	fset := token.NewFileSet()
+	dir := flag.String("dir", "", "project directory")
```

Comment:
`s/project directory/package directory/` I do think that we expect single package inside one directory that is being checked. That's the most common case.

Reply:
I also thank about warning type that will let us write logs in different formats.